### PR TITLE
Closes #32 advs template growth parameters

### LINKED ISSUE #32: ADVS template for growth parameters
See the outline document in MS Teams area. Steps 5 and 6 rely on the new functions being available so placeholders will be needed for now. 

The main part will be Step 4 for preparing the metadata into a consistent structure covering the below points:

**PROPOSAL FOR BY AGE:** In this step for each of BMI/HEIGHT/WEIGHT, we would combine WHO <2yrs and CDC >=2 data into one single consistently structured meta dataset (with `SEX` as a variable - with `M` and `F` values, `AGE` in days via a conversion factor). Clearly comment to make clear here our default reference source, in our case WHO <2 yrs and CDC >=2, and the fact that any user could replace this with their own chosen metadata. We can explain this then more in the vignette. 

**PROPOSAL FOR THE WHO-ONLY DATA:** In this step for each of HEAD CIRCUMFERENCE and WEIGHT (BY LENGTH/HEIGHT), we would combine the boys and girls data into one single consistently structured meta dataset (with `SEX` as a variable - with `M` and `F` values). For WEIGHT (BY LENGTH/HEIGHT) additionally then combine the LENGTH and HEIGHT data into one single dataset with additional variable `MEASURE` to denote LENGTH or HEIGHT.  

The naming of these meta datasets should still clearly convey which parameter they each relate to, so then no need to add PARAMCD here as a variable in these datasets.

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by Fanny-Gautier** [comment]
Awaiting for #34, which needs to be implemented
Awaiting for #36 to implement ADVS template with `adsl_peds`

File: inst/templates/ad_advs.R [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    )
+  ) %>%
+  derive_vars_merged(
+    dataset_add = windows,
+    by_vars = exprs(AVISIT)
+  )
+
+## Add PARAM/PARAMN ----
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:237]: wouldn't you add this after the newly derived records are created?

File: inst/templates/ad_advs.R [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+) %>%
+  mutate(
+    PARAMN = case_when(
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:263]: See comments above for PARAMN

File: inst/templates/ad_advs.R [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      PARAMCD == "WTASDS" ~ 5,
+      PARAMCD == "WTAPCTL" ~ 6,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("WTASDS", "WTAPCTL"))
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:269]: i dont think needed as fine to keep source params too

File: inst/templates/ad_advs.R [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+# )  %>%
+# filter(PARAMCD %in% c("WGTHSDS", "WGTHPCTL"))
+
+## Combine all derived parameters together
+advs <- advs %>%
+  rbind(advs_wgt_age, advs_bmi_age, advs_hdc_age)
+# z-score and percentile for HEIGHT and LENGTH to be added once the
+# `derive_params_growth_height` function is ready
+
+## Get ASEQ ----
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:355]: Also derive ANL01FL?

File: inst/templates/ad_advs.R [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+  mutate(
+    PARAMN = case_when(
+      PARAMCD == "HDCSDS" ~ 9,
+      PARAMCD == "HDCPCTL" ~ 10,
+      TRUE ~ PARAMN
+    )
+  ) %>%
+  filter(PARAMCD %in% c("HDCSDS", "HDCPCTL"))
+
+## Calculate for Height/Length ----
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:325]: also for this step Samia left a note in the outline doc: For-Height/Length SDS/Percentiles
Create a temporary Height variable in cm (e.g. HGTTMP) and add the height/length at each time point

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
Thanks @Fanny-Gautier and @Lina2689 - here's some early review comments for consideration. I'll be able to test further once the functions are ready and then i can try out the code too. I also updated the MS teams outline doc for your comments - thanks for these! It's coming together great and nice to be able to visualize this code altogether now.

File: inst/templates/ad_advs.R [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+  ) %>%
+  mutate(BRTHDT = convert_dtc_to_dt(
+    BRTHDTC,
+    highest_imputation = "n",
+    date_imputation = "first",
+    min_dates = NULL,
+    max_dates = NULL,
+    preserve = FALSE
+  )) %>%
+  ## Calculate ADT ----
```
[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R:182]: It needs ADSL.TRTSTDT. Corresponding Pediatric ADSL is currently missing.

File: inst/templates/ad_advs.R (hunk 4 above)
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R:355]: How can we know which records are post-baseline without Treatment start date ? Currently no Pediatric ADSL available.

File: inst/templates/ad_advs.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:182]: ah yes, we can add it later then, thanks!

File: inst/templates/ad_advs.R (hunk 4 above)
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:355]: ok, we can add later, thanks! i was just comparing vs the admiral ADVS template to see if we missed anything

File: inst/templates/ad_advs.R (hunk 3 above)
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:269]: See from line 105 in https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_advs.R - we usually just build up the ADaM by taking VS records and then adding one new derived parameter each time by piping to the next `derive_param_xxx()` function. Our `parameter = VSTESTCD == "WEIGHT"` argument for example would choose which input records to use for the derived parameter each time. Does that make sense?

File: inst/templates/ad_advs.R [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+#
+# Input: adsl, vs
+#        WHO_bmi_for_age_boys, WHO_bmi_for_age_girls, cdc_bmiage,
+#        who_lgth_ht_for_age_boys, who_lgth_ht_for_age_girls, cdc_htage,
+#        who_wt_for_age_boys, who_wt_for_age_girls, cdc_wtage,
+#        who_hc_for_age_boys, who_hc_for_age_girls,
+#        who_wt_for_ht_boys, who_wt_for_ht_girls, who_wt_for_lgth_boys, who_wt_for_lgth_girls
+
+library(admiral)
+library(admiraldev)
```
[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R]: admiraldev currently needed for `derive_params_growth_age()`
image

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: did you run `devtools::load_all()` first? i dont think its needed because we import the required functions from {admiraldev} as part of the package so you don't need the users to install and load all of admiraldev which is very heavy. this was the same comment i had for {rlang} as i don't think you need to include either here. neither of these are included in any of the admiral core templates for example.

[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R]: Not today, I forgot. Done now, and I'll think about in the future. Thank you for the reminder.

**Comment by Lina2689** [comment]
@rossfarrugia PR is ready for review. All comments are addressed except the one related to ADSL dataset.

**Comment by rossfarrugia** [comment]
> @rossfarrugia PR is ready for review. All comments are addressed except the one related to ADSL dataset.

thanks Lina for all the updates! i've been working through the first draft of the vignette for now, so will re-review this PR later once we get the others ready that this relies on like the ADSL and the 2 functions.

**Comment by Lina2689** [comment]
> @Fanny-Gautier @Lina2689 the PRs for the updated `dm_peds`, `vs_peds` and `adsl_peds` test data are now taking shape (see #39 from @remikk and #43 from me as FYI).

@rossfarrugia Updated the template to utilize peds test datasets. 

Everything is completed from our end except the function from #34, which is not ready yet.

**Comment by Fanny-Gautier** [comment]
> Thanks @Lina2689! you can remove mention of `dm_peds` from this one now. once both the functions are stable then i can do some more testing here and trying out running all the code.

Done. Thanks.

File: inst/templates/ad_advs.R [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+# https://pharmaverse.github.io/admiral/articles/admiral.html#handling-of-missing-values # nolint
+
+vs <- convert_blanks_to_na(vs)
+
+# Lookup tables ----
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 1,
+  "HEIGHT", "HEIGHT", "Height (cm)", 2,
```
Comment (Minlei0201) [inline inst/templates/ad_advs.R:162]: Should LENGTH also be included in the lookup table?

File: inst/templates/ad_advs.R [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT,
+    out_unit = "DAYS",
+    trunc_out = FALSE
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
```
Comment (Minlei0201) [inline inst/templates/ad_advs.R:211]: Is there any reason why PARAM is not added at the same time with PARAMCD?

Comment (Minlei0201) [reply inst/templates/ad_advs.R:211]: Nvm, I see PARAM being added after the z-scores and percentiles being derived in line 335, which do not have VSTESTCD.

File: inst/templates/ad_advs.R (hunk 8 above)
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:162]: @Minlei0201 in our template (as per the outline doc we shared on our admiralpeds MS teams) we only use the HEIGHT parameter here as default but in the vignette we'll explain around body length. Most CRFs only likely collect as height in one parameter and then it depends on the child's age what they actually collect here.

**Comment by Fanny-Gautier** [comment]
Hi @rossfarrugia,
we have implemented derive_params_growth_height function. Could you please review the DRAFT PR and let us know if there any comments? Thanks.

**Comment by rossfarrugia** [comment]
Sure @Fanny-Gautier - will try and take a look tomorrow. Thanks to you and Lina for the quick updates following Zelos's work on the latest new function.

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
@Fanny-Gautier @Lina2689 thanks for all the updates! i took another look today and here's some more comments i spotted. let me know if any are unclear. once Zelos's function updates are made and the corresponding updates here, then happy to do a final review and test run everything together.

File: inst/templates/ad_advs.R [hunk 10]
```diff
@@ -0,0 +1,552 @@
...
+who_hc_for_age <- who_hc_for_age_boys %>%
+  mutate(SEX = "M") %>%
+  bind_rows(who_hc_for_age_girls %>%
+    mutate(SEX = "F")) %>%
+  rename(AGE = Day) %>%
+  # AGEU is added in metadata, required for derive_params_growth_age
+  mutate(AGEU = "DAYS") %>%
+  arrange(AGE, SEX)
+
+## WHO - WEIGHT for HEIGHT/HEIGHT ----
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:118]:
```suggestion
## WHO - WEIGHT for LENGTH/HEIGHT ----
```

File: inst/templates/ad_advs.R [hunk 11]
```diff
@@ -0,0 +1,552 @@
+# Name: ADVS
+#
+# Label: Vital Signs Analysis Dataset for Pediatric Trials
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:3]:
```suggestion
# Label: Vital Signs Analysis Dataset including Anthropometric indicators for Pediatric Trials
```

File: inst/templates/ad_advs.R [hunk 12]
```diff
@@ -0,0 +1,552 @@
...
+    PARAMCD = "WTASDS",
+    PARAM = "Weight-for-age z-score"
+  ),
+  set_values_to_pctl = exprs(
+    PARAMCD = "WTAPCTL",
+    PARAM = "Weight-for-age percentile"
+  )
+)
+
+## Calculate BMI for AGE z-score and Percentile ----
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:283]: Add a section title such as "Derive Anthropometric indicators parameters" above this section. I see Zelos has been using that wording in his function headers

File: inst/templates/ad_advs.R [hunk 13]
```diff
@@ -0,0 +1,552 @@
...
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+
+# Add Current HEIGHT Temporary variable (in cm)
+## Calculate Current HEIGHT/LENGTH at each time point Temporary variable----
+hgttmp <- advs %>%
+  filter((PARAMCD == "HEIGHT" | PARAMCD == "LENGTH") & VSSTRESU == "cm") %>%
+  select(STUDYID, USUBJID, HGTTMP = AVAL, HGTTMPU = VSSTRESU, AVISIT)
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:254]: there's cases in this template of STUDYID, USUBJID hard-coded, but we should use `get_admiral_option("subject_keys")` so this is flexible and can be changed from the default by the user in one place. see this admiral core PR for example for the updates needed: https://github.com/pharmaverse/admiral/pull/2452/files

File: inst/templates/ad_advs.R [hunk 14]
```diff
@@ -0,0 +1,552 @@
...
+# Label: Vital Signs Analysis Dataset for Pediatric Trials
+#
+# Input: adsl, vs
+#        WHO_bmi_for_age_boys, WHO_bmi_for_age_girls, cdc_bmiage,
+#        who_lgth_ht_for_age_boys, who_lgth_ht_for_age_girls, cdc_htage,
+#        who_wt_for_age_boys, who_wt_for_age_girls, cdc_wtage,
+#        who_hc_for_age_boys, who_hc_for_age_girls,
+#        who_wt_for_ht_boys, who_wt_for_ht_girls, who_wt_for_lgth_boys, who_wt_for_lgth_girls
+
+library(admiral)
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R:12]: @rossfarrugia if we add library(admiralpeds), only vs_peds data is getting updated with below USUBJIDs; however adsl_peds has different set of subjects. Due to this, we are unable to merge vs_peds and ADSL_peds. Please refer below snapshots and let us know if ADSL_peds data will also get updated. 

**vs_peds:**
image

**ADSL_peds:**
image

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:12]: p.s. you don't need to run `library(admiralpeds)` yourself as you can use `devtools::load_all()` here as you're working within this package. the `library(admiralpeds)` is for our users so that they can first load the package and get access to all our new functions/test data/metadata/etc

File: inst/templates/ad_advs.R [hunk 15]
```diff
@@ -0,0 +1,552 @@
...
+  set_values_to_pctl = exprs(
+    PARAMCD = "HDCPCTL",
+    PARAM = "HDC-for-age percentile"
+  )
+)
+
+## Calculate Z-Score and percentile parameters for Height/Length ----
+message("To derive height/length parameters, below function needs to call separately
+        for Height and Length based on the input data and current age of the patient,
+        as it depends on your CRF guidelines.")
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R:324]: @rossfarrugia We can update ADVS peds template once `derive_params_growth_height()` is updated accordingly.

File: inst/templates/ad_advs.R [hunk 16]
```diff
@@ -0,0 +1,552 @@
...
+) %>%
+  filter(PARAMCD %in% c("HDCHSDS", "HDCHPCTL"))
+
+## Combine the records for Weight, BMI and Head Circumference for Height/Length ----
+advs <- advs_age %>%
+  bind_rows(
+    advs_lgth %>% filter(PARAMCD %in% c("WGTHSDS", "WGTHPCTL")),
+    advs_ht %>% filter(PARAMCD %in% c("WGTHSDS", "WGTHPCTL")),
+    BMI_lgth, BMI_ht, HDC_lgth, HDC_ht
+  )
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R:452]: @rossfarrugia will be modified once height function is updated.

File: inst/templates/ad_advs.R [hunk 17]
```diff
@@ -0,0 +1,552 @@
...
+  derive_var_base(
+    by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+    source_var = AVAL,
+    new_var = BASE
+  ) %>%
+  # Calculate CHG
+  derive_var_chg() %>%
+  # Calculate PCHG
+  derive_var_pchg() %>%
+  # Keeping CHG/PCHG blank for pre-treatment(Screening/BASELINE) records
```
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R:508]: @rossfarrugia Was this comment regarding line #508 (i.e. # Keeping CHG/PCHG blank for pre-treatment(Screening/BASELINE) records) ?

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:508]: yes - i think we should remove this and just match the admiral core ADVS template here or users won't understand why the different conventions between the 2 packages. then if users want to only show for post-baseline (which definitely makes most sense to me!) then they could. 

but also they wouldn't need the extra steps you added here, e.g. for admiralroche we use:

```
  # derive CHG
  restrict_derivation(
    derivation = derive_var_chg,
    filter = !(AVISIT %in% c("BASELINE", "PRE-BASELINE"))
  ) %>%
  # derive PCHG
  restrict_derivation(
    derivation = derive_var_pchg,
    filter = !(AVISIT %in% c("BASELINE", "PRE-BASELINE"))
  )
```

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:406]: either that or we also update every admiral core BDS template that includes CHG/PCHG with the above extract? my main comment is just to make them consistent either way

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: Yes, i'm not sure why similar was never included in admiral core templates. there must have been some discussion on this and maybe some companies do populated for all records, but i would always have thought same as you both. thanks for updating at least for consistency with admiral core here and we can leave it to users.

File: inst/templates/ad_advs.R [hunk 18]
```diff
@@ -167,18 +168,16 @@ param_lookup <- tibble::tribble(
...
-  NA_character_, "HDCHSDS", "HDC-for-length/height Z-Score", 15,
-  NA_character_, "HDCHPCTL", "HDC-for-length/height Percentile", 16
+  NA_character_, "WGTASDS", "Weight-for-age z-score", 5,
+  NA_character_, "WGTAPCTL", "Weight-for-age percentile", 6,
+  NA_character_, "HGTSDS", "Height-for-age z-score", 7,
+  NA_character_, "HGTPCTL", "Height-for-age percentile", 8,
+  NA_character_, "BMISDS", "BMI-for-age z-score", 9,
+  NA_character_, "BMIPCTL", "BMI-for-age percentile", 10,
+  NA_character_, "HDCSDS", "HDC-for-age z-score", 11,
+  NA_character_, "HDCPCTL", "HDC-for-age percentile", 12,
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:178]: can we have the PARAM for these as "Head Circumference-for-age..." instead of "HDC-for-age" which would not be as meaningful to for example an FDA reviewer.

File: inst/templates/ad_advs.R [hunk 19]
```diff
@@ -247,83 +241,90 @@ advs <- advs %>%
...
-  select(STUDYID, USUBJID, HGTTMP = AVAL, HGTTMPU = VSSTRESU, AVISIT)
-
+# Derive Current HEIGHT/LENGTH at each time point Temporary variable----
 advs <- advs %>%
   derive_vars_merged(
-    dataset_add = hgttmp,
-    by_vars = exprs(STUDYID, USUBJID, AVISIT)
+    dataset_add = advs,
+    by_vars = c(get_admiral_option("subject_keys"), exprs(AVISIT)),
+    filter_add = PARAMCD == "HEIGHT" & VSSTRESU == "cm",
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:249]: Should we use `toupper()` here and "CM"?  as might be a bit more robust as i imagine most companies map these in uppercase

File: inst/templates/ad_advs.R [hunk 20]
```diff
@@ -517,26 +415,45 @@ advs <- advs %>%
...
+    filter_pre_timepoint = AVISIT == "Baseline"
+  )
+
+## ANL01FL: Flag last result within an AVISIT and ATPT for post-baseline records ----
+advs <- advs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      new_var = ANL01FL,
+      by_vars = exprs(USUBJID, PARAMCD, AVISIT, ATPT),
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:422]:
```suggestion
      by_vars = c(get_admiral_option("subject_keys"), exprs(PARAMCD, AVISIT, ATPT)),
```

File: inst/templates/ad_advs.R (hunk 18 above)
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R]: implemented

File: inst/templates/ad_advs.R (hunk 19 above)
Comment (Fanny-Gautier) [reply inst/templates/ad_advs.R]: implemented

File: inst/templates/ad_advs.R (hunk 15 above)
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R]: @rossfarrugia We can keep it as is, as it was discussed to make separate calls for HEIGHT and LENGTH.

File: inst/templates/ad_advs.R (hunk 16 above)
[the remark this answers is not in the document]
Comment (Lina2689) [reply inst/templates/ad_advs.R]: @rossfarrugia Updated to use pipe to add new parameters for HEIGHT/LENGTH.

File: inst/templates/ad_advs.R (hunk 15 above)
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: agree - only i would then restructure a bit, e.g. currently you combine all this metadata as one and then split it out again via `who_wt_for_ht_lgth %>% filter(MEASURE == "LENGTH")`. this doesn't make sense to me as you could just make 1 `who_wt_for_ht` and 1 `who_wt_for_lgth` instead at the top metadata section. and you dont even need a `HEIGHT_LENGTH` variable there anymore for example.

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: you mean the template draft outline doc on our MS teams? thats just a starter to get us building these issues/PRs. your PR is far more advanced than that now so i've not been maintaining that working word doc, e.g. it still talks of doing these parameters by one function call whereas Zelos found better to run the function twice. we can leave that doc now and instead focus on the latest in our PRs as the latest plan. we don't really maintain "specs" in same way as you do when stats programming here, its mainly just the GH issue and then the PR review discussoins that guide us

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: and ultimatelty the vignette will then explain everything to users in the end

[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_advs.R]: sorry @Lina2689 @Fanny-Gautier at the team call today i realised i had incorrectly said above "you dont even need a HEIGHT_LENGTH variable" - i meant `MEASURE` here as we don't need any variable to say "Height" or "Length" if we no longer join these into one metadata file. We still should make the values and units consistent so that they work with the function from @zdz2101 (i.e. `HEIGHT_LENGTH` and `HEIGHT_LENGTHU`)

File: inst/templates/ad_advs.R (hunk 8 above)
Comment (Minlei0201) [reply inst/templates/ad_advs.R:167]: I see, thanks @rossfarrugia ! So if a study collects length, it will be bind with the height, and mapped to `PARAMCD="HEIGHT`"?

Comment (rossfarrugia) [reply inst/templates/ad_advs.R:167]: for the sake of our test data and examples yes, but whatever the study mapped this as they can feed it into the functions using the `parameter` argument so the user has complete flexibility here.

File: inst/templates/ad_advs.R [hunk 21]
```diff
@@ -0,0 +1,452 @@
...
+    set_values_to_sds = exprs(
+      PARAMCD = "HGTSDS",
+      PARAM = "Height-for-age z-score"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "HGTPCTL",
+      PARAM = "Height-for-age percentile"
+    )
+  ) %>%
+  derive_params_growth_age(
```
Comment (rossfarrugia) [inline inst/templates/ad_advs.R:288]: For BMI only i think we should set `bmi_cdc_correction = TRUE` as David had advised us that it is appropriate to use the CDC developed extended percentiles (>95%) as default to monitor high BMI values

**Comment by rossfarrugia** [comment]
@Lina2689 the 2 functions are now merged in main, so you should be able to merge these here now and then mark this one as ready for review if you're happy with it. once you are then add "closes" at start of the PR title so it auto-closes the issue later and put a message out on our slack dev channel to let the team know to review this. i'll do final checks this week from my side.

File: inst/templates/ad_advs.R (hunk 21 above)
Comment (rossfarrugia) [reply inst/templates/ad_advs.R:288]: @Lina2689 FYI to update this one too

Comment (Lina2689) [reply inst/templates/ad_advs.R:288]: @zdz2101 @rossfarrugia I set `bmi_cdc_correction = TRUE` for BMI, but its throwing an error for P95 and sigma. Can you please suggest from I could get this?

image

Comment (rossfarrugia) [reply inst/templates/ad_advs.R:286]: @Lina2689 are you sure you're using the right source metadata? maybe you didn't merge the latest `main` version of https://github.com/pharmaverse/admiralpeds/blob/main/data-raw/cdc_bmi.R here. if you look at that file it does have those columns. or if you spot an issue here with `data/cdc_bmiage.rda` let us know.

Comment (rossfarrugia) [reply inst/templates/ad_advs.R:286]: maybe your `bmi_for_age` step of preprocessing is inadvertently dropping these variables.

Comment (Lina2689) [reply inst/templates/ad_advs.R:286]: Thanks @rossfarrugia I missed to refresh the cdc_bmiage.rda file.

**Review by zdz2101 — APPROVED** [review]
LGTM!

**Comment by rossfarrugia** [comment]
@Lina2689 @zdz2101 i tested the full template and it runs well now as far as i can tell. a few review questions:

1) why do some patients have `BMISDS` as `NA` for all visits, e.g. `01-701-1015`?

2) is it correct that some patients have wildly low or high pecentile values, e.g. one even has 100% for one parameter i spotted? i had remembered Remi tried to make all the test data values realistic depending on age so was surprised to see this.

3) @zdz2101 i had to make a minor commit to the template to drop variables `P95` and `Sigma` - shouldn't these really be dropped as part of the function already depending on whether the user set `bmi_cdc_correction = TRUE`? If so feel free to correct the function here in this PR and remove my one liner from the template and we can approve/merge both in one go.

**Comment by zdz2101** [comment]
> @Lina2689 @zdz2101 i tested the full template and it runs well now as far as i can tell. a few review questions:
> 
> 1. why do some patients have `BMISDS` as `NA` for all visits, e.g. `01-701-1015`?
> 2. is it correct that some patients have wildly low or high pecentile values, e.g. one even has 100% for one parameter i spotted? i had remembered Remi tried to make all the test data values realistic depending on age so was surprised to see this.
> 3. @zdz2101 i had to make a minor commit to the template to drop variables `P95` and `Sigma` - shouldn't these really be dropped as part of the function already depending on whether the user set `bmi_cdc_correction = TRUE`? If so feel free to correct the function here in this PR and remove my one liner from the template and we can approve/merge both in one go.

Per 1) Looks like there was a bug in the BMI correction when it comes to certain age joins, my apologies for the negligence @Lina2689 @rossfarrugia @Fanny-Gautier

**Comment by rossfarrugia** [comment]
@zdz2101 i'm going to approve and merge this one now but can you drop David an email to sanity check the wildly low percentile values to see he agrees? e.g. one patient even has negative values here - wouldn't this just get set as zero? 🤷‍♂️

**Review by rossfarrugia — APPROVED** [review]
